### PR TITLE
Ei 963 application complete

### PR DESCRIPTION
**Pull request by David-B-Read**

### CODE REVIEW

File: src/SFA.DAS.EmployerIncentives.Web/Services/BankDetails/IBankDetailsStatusService.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Threading.Tasks;
+using SFA.DAS.EmployerIncentives.Web.Models;
+
+namespace SFA.DAS.EmployerIncentives.Web.Services.BankDetails
+{
+    public interface IBankDetailsStatusService
+    {
+        Task<LegalEntityModel> RecordBankDetailsComplete(string accountId, string accountLegalEntityId);
```
Comment (Stephek) [inline src/SFA.DAS.EmployerIncentives.Web/Services/BankDetails/IBankDetailsStatusService.cs:8]: Why would a method that updates return an unrelated object?  This doesn't follow the command and query separation pattern.  

Do we need a bank details status service?  This seems very granular, I would expect it to act on a BankDetailsStatus object.  There's already a bank details service and an application service so wouldn't it be better on one of those interfaces?  Isn't the status change against the application?  So would be better on that interface perhaps?

**Review by Stephek — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.EmployerIncentives.Web/Services/BankDetails/IBankDetailsStatusService.cs (hunk 1 above)
Comment (David-B-Read) [reply src/SFA.DAS.EmployerIncentives.Web/Services/BankDetails/IBankDetailsStatusService.cs:8]: The status is recorded against the legal entity, so will move into LegalEntitiesService

File: src/SFA.DAS.EmployerIncentives.Web/Controllers/ApplicationCompleteController.cs [hunk 2]
```diff
@@ -1,36 +1,37 @@
...
         }
 
         [HttpGet]
         [Route("")]
         public async Task<IActionResult> Confirmation(string accountId, Guid applicationId)
         {
             var application = await _applicationService.Get(accountId, applicationId, includeApprenticeships: false);
-            var legalEntityName = await GetLegalEntityName(accountId, application.AccountLegalEntityId);
-            var model = new ConfirmationViewModel(accountId, application.AccountLegalEntityId, legalEntityName, _configuration.ManageApprenticeshipSiteUrl);
+            var legalEntity = await _bankDetailsStatusService.RecordBankDetailsComplete(accountId, application.AccountLegalEntityId);
```
Comment (ruslanss) [inline src/SFA.DAS.EmployerIncentives.Web/Controllers/ApplicationCompleteController.cs]: this doesn't feel quite right

Comment (ruslanss) [reply src/SFA.DAS.EmployerIncentives.Web/Controllers/ApplicationCompleteController.cs]: RecordBankDetailsComplete sounds like a command, so shouldn't return anything. can LE be fetched separately after completing the command?

Comment (David-B-Read) [reply src/SFA.DAS.EmployerIncentives.Web/Controllers/ApplicationCompleteController.cs]: That's been changed in the latest commit

**Review by Stephek — APPROVED** [review]

**Pull request merged**